building bevy_tasks v0.2.1 fails on wasm32 target

```
$ cargo build --example headless_wasm --target wasm32-unknown-unknown --no-default-features

   Compiling bevy_tasks v0.2.1 (/home/mrk/private/rust/bevy/crates/bevy_tasks)
warning: unused imports: `mem`, `pin::Pin`
 --> /home/mrk/private/rust/bevy/crates/bevy_tasks/src/single_threaded_task_pool.rs:3:5
  |
3 |     mem,
  |     ^^^
4 |     pin::Pin,
  |     ^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0597]: `executor` does not live long enough
  --> /home/mrk/private/rust/bevy/crates/bevy_tasks/src/single_threaded_task_pool.rs:69:23
   |
61 |     pub fn scope<'scope, F, T>(&self, f: F) -> Vec<T>
   |                  ------ lifetime `'scope` defined here
...
69 |             executor: &executor,
   |                       ^^^^^^^^^ borrowed value does not live long enough
...
73 |         f(&mut scope);
   |         ------------- argument requires that `executor` is borrowed for `'scope`
...
83 |     }
   |     - `executor` dropped here while still borrowed

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0597`.
error: could not compile `bevy_tasks`.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -55,6 +55,13 @@ jobs:
           CARGO_INCREMENTAL: 0
           RUSTFLAGS: "-C debuginfo=0 -D warnings"
 
+      - name: Check wasm
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          target: wasm32-unknown-unknown
+          override: true
+
   clean:
     runs-on: ubuntu-latest
     steps:

--- a/crates/bevy_tasks/src/single_threaded_task_pool.rs
+++ b/crates/bevy_tasks/src/single_threaded_task_pool.rs
@@ -1,7 +1,6 @@
 use std::{
     future::Future,
     mem,
-    pin::Pin,
     sync::{Arc, Mutex},
 };
 
@@ -63,10 +62,12 @@ impl TaskPool {
         F: FnOnce(&mut Scope<'scope, T>) + 'scope + Send,
         T: Send + 'static,
     {
-        let executor = async_executor::LocalExecutor::new();
+        let executor = &async_executor::LocalExecutor::new();
+        let executor: &'scope async_executor::LocalExecutor<'scope> =
+            unsafe { mem::transmute(executor) };
 
         let mut scope = Scope {
-            executor: &executor,
+            executor,
             results: Vec::new(),
         };
 

